source/include/azure_iot.h

```diff
@@ -0,0 +1,153 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file azure_iot.h
+ *
+ * @brief Azure IoT FreeRTOS middleware common APIs and structs
+ *
+ * @note You MUST NOT use any symbols (macros, functions, structures, enums, etc.)
+ * prefixed with an underscore ('_') directly in your application code. These symbols
+ * are part of Azure SDK's internal implementation; we do not document these symbols
+ * and they are subject to change in future versions of the SDK which would break your code.
+ *
+ */
+
+#ifndef AZURE_IOT_H
+#define AZURE_IOT_H
+
+/* AZURE_IOT_DO_NOT_USE_CUSTOM_CONFIG allows building the azure iot library
+ * without a custom config. If a custom config is provided, the
+ * AZURE_IOT_DO_NOT_USE_CUSTOM_CONFIG macro should not be defined. */
+#ifndef AZURE_IOT_DO_NOT_USE_CUSTOM_CONFIG
+    /* Include custom config file before other headers. */
+    #include "azure_iot_config.h"
+#endif
+
+/* Include config defaults header to get default values of configs not
+ * defined in azure_iot_mqtt_config.h file. */
+#include "azure_iot_config_defaults.h"
+
+#include "FreeRTOS.h"
+
+#include "azure/az_iot.h"
+
+typedef enum AzureIoTResult
+{
+    eAzureIoTSuccess = 0,     /**< Success. */
+    eAzureIoTInvalidArgument, /**< Input argument does not comply with the expected range of values. */
+    eAzureIoTOutOfMemory,     /**< The system is out of memory. */
+    eAzureIoTItemNotFound,    /**< The item was not found. */
+    eAzureIoTFailed,          /**< There was a failure. */
+} AzureIoTResult_t;
+
+/**
+ * @brief The bag of properties associated with a message.
+ *
+ */
+typedef struct AzureIoTMessageProperties
+{
+    struct
+    {
+        az_iot_message_properties xProperties;
+    } _internal;
+} AzureIoTMessageProperties_t;
+
+typedef uint64_t ( * AzureIoTGetCurrentTimeFunc_t )( void );
+
+typedef uint32_t ( * AzureIoTGetHMACFunc_t )( const uint8_t * pucKey,
+                                              uint32_t ulKeyLength,
+                                              const uint8_t * pucData,
+                                              uint32_t ulDataLength,
+                                              uint8_t * pucOutput,
+                                              uint32_t ulOutputLength,
+                                              uint32_t * pulBytesCopied );
+
+/**
+ * @brief Initialize Azure IoT middleware.
```

Comment:
I'm assuming this should be called once per process, correct?  (If process is even right word for RTOS, which I think it is even though there's just the 1.)  This singletnon-ness of the API even though seemingly obivous I think worth documenting.

Reply:
good call noted